### PR TITLE
tidb-in-kubernetes: add reference for tkctl

### PR DESCRIPTION
**Pull request by aylei**
Preview: https://github.com/aylei/docs-cn/blob/tkctl/tidb-in-kubernetes/tkctl.md

This documentation has a English version which is not properly organized, I'll commit that to `pingcap/docs` later, namely, this documentation do not need translation😜

@tennix @xiaojingchen @dcalvin @lilin90 PTAL

Signed-off-by: Aylei <rayingecho@gmail.com>

### CODE REVIEW

File: tidb-in-kubernetes/tkctl.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+$ tkctl debug demo-cluster-tikv-0
+$ cp /proc/1/root/tikv-server /
+$ ./run_flamegraph.sh 1
+```
+
+### tkctl ctop
+
+`tkctl ctop [pod_name | node/node_name ]`
+
+该命令用于查看集群中 Pod 或 Node 的实时监控信息，相比于 `kubectl top`，`tkctl ctop` 还会展示网络和磁盘的使用信息。
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
该命令用于查看集群中 Pod 或 Node 的实时监控信息，和 `kubectl top` 相比，`tkctl ctop` 还会展示网络和磁盘的使用信息。
```

File: tidb-in-kubernetes/tkctl.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+```shell
+$ tkctl debug demo-cluster-tikv-0
+$ cp /proc/1/root/tikv-server /
+$ ./run_flamegraph.sh 1
+```
+
+### tkctl ctop
+
+`tkctl ctop [pod_name | node/node_name ]`
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
命令的完整形式：`tkctl ctop [pod_name | node/node_name ]`。
```

File: tidb-in-kubernetes/tkctl.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+例子：
+
+```
+$ tkctl help debug
+```
+
+### tkctl options
+
+该命令用于展示 `tkctl` 的所有的全局参数：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
该命令用于展示 `tkctl` 的所有的全局参数。
```

File: tidb-in-kubernetes/tkctl.md [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
例子：

```

File: tidb-in-kubernetes/tkctl.md [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+  -t, --tidbcluster='': Tidb cluster name
+      --token='': Bearer token for authentication to the API server
+      --user='': The name of the kubeconfig user to use
+  -v, --v=0: log level for V logs
+      --vmodule=: comma-separated list of pattern=N settings for file-filtered logging
+```
+
+这些参数主要用于指定如何连接 Kubernetes 集群，其中最常用的参数是： 
+
+- `--context`: 指定目标 Kubernetes 集群
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
- `--context`：指定目标 Kubernetes 集群
```

File: tidb-in-kubernetes/tkctl.md [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+- `--namespace`: 指定 Namespace
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
- `--namespace`：指定 Namespace
```

File: tidb-in-kubernetes/tkctl.md [hunk 7]
```diff
@@ -0,0 +1,282 @@
+---
+title: tkctl 使用说明
+category: reference
+---
+
+# tkctl 使用说明
+
+tkctl(TiDB Kubernetes Controltkctl) 为 TiDB in Kubernetes 设计的命令行工具，用于运维集群和诊断集群问题。
```
[the remark this answers is not in the document]
Comment (aylei) [reply tidb-in-kubernetes/tkctl.md]: 是的，Controltkctl 是一个 typo

**Comment by aylei** [comment]
@yikeke All comments are addressed, PTAL again

File: tidb-in-kubernetes/tkctl.md [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+### tkctl list
+
+该命令用于列出所有已安装的 TiDB 集群：
+
+| 参数             | 缩写      | 说明                                                       |
+| -----            | --------- | -----------                                                |
+| --all-namespaces | -A        | 是否查询所有的 Kubernetes Namespace                        |
+| --output         | -o        | 输出格式，可选值有 [default,json,yaml]，默认值为 `default` |
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
示例如下：
```

File: tidb-in-kubernetes/tkctl.md [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+NAMESPACE NAME           PD    TIKV   TIDB   AGE
+foo       demo-cluster   3/3   3/3    2/2    11m
+bar       demo-cluster   3/3   3/3    1/2    11m
+```
+
+### tkctl use
+
+该命令用于指定当前 `tkctl` 操作的 TiDB 集群，在使用该命令设置当前操作的 TiDB 集群后，所有针对集群的操作命令会自动选定该集群，从而可以略去 `--tidbcluster` 参数。
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
示例如下：
```

File: tidb-in-kubernetes/tkctl.md [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+
+### tkctl info
+
+该命令用于展示 TiDB 集群的信息。
+
+| 参数           | 缩写      | 说明                                       |
+| -----          | --------- | -----------                                |
+| --tidb-cluster | -t        | 指定 TiDB 集群，默认为当前使用的 TiDB 集群 |
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
示例如下：
```

File: tidb-in-kubernetes/tkctl.md [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+该命令用于获取 TiDB 集群中组件的详细信息。
+
+可选的组件 (`component`) 有： `pd`、`tikv`、`tidb`、`volume` 和 `all`（用于同时查询所有组件）。
+
+| 参数 | 缩写 | 说明 |
+| ----- | --------- | ----------- |
+| --tidb-cluster | -t | 指定 TiDB 集群，默认为当前使用的 TiDB 集群 |
+| --output | -o | 输出格式，可选值有 `default`、`json` 和 `yaml`，默认值为 `default` |
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
示例如下：
```

File: tidb-in-kubernetes/tkctl.md [hunk 12]
```diff
@@ -0,0 +1,288 @@
...
+| --image         |           | 指定 debug 容器使用的镜像，默认为 `pingcap/tidb-debug:lastest`                                                                  |
+| --container     | -c        | 选择需要诊断的容器，默认为 Pod 定义中的第一个容器                                                                               |
+| --docker-socket |           | 指定目标节点上的 Docker Socket，默认为 `/var/run/docker.sock`                                                                   |
+| --privileged    |           | 是否为 debug 容器开启 [privileged](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities) 模式 |
+
+> **注意：**
+>
+> Debug 容器使用的默认镜像包含了绝大多数的诊断工具，因此体积较大，假如只需要 `pd-ctl` 和 `tidb-ctl`，可以使用 `--image=pingcap/tidb-control:latest` 来指定使用 `tidb-control` 镜像。
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
示例如下：
```

File: tidb-in-kubernetes/tkctl.md [hunk 13]
```diff
@@ -0,0 +1,288 @@
...
+命令的完整形式：`tkctl ctop [pod_name | node/node_name ]`。
+
+该命令用于查看集群中 Pod 或 Node 的实时监控信息，和 `kubectl top` 相比，`tkctl ctop` 还会展示网络和磁盘的使用信息。
+
+| 参数            | 简写      | 描述                                                                         |
+| -----           | --------- | -----------                                                                  |
+| --image         |           | 指定 ctop 的镜像，默认为 `quay.io/vektorlab/ctop:0.7.2` |
+| --docker-socket |           | 指定 ctop 使用的 Docker Socket，默认为 `/var/run/docker.sock` |
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
示例如下：
```

File: tidb-in-kubernetes/tkctl.md [hunk 14]
```diff
@@ -0,0 +1,288 @@
...
+```
+$ tkctl ctop demo-cluster-tikv-0
+$ tkctl ctop node/172.16.4.155
+```
+
+### tkctl help [command]
+
+该命令用于展示各个子命令的帮助信息。
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
示例如下：
```

File: tidb-in-kubernetes/tkctl.md [hunk 15]
```diff
@@ -0,0 +1,288 @@
...
+
+```
+$ tkctl help debug
+```
+
+### tkctl options
+
+该命令用于展示 `tkctl` 的所有的全局参数。
+
+例子：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
示例如下：
```

File: tidb-in-kubernetes/tkctl.md [hunk 16]
```diff
@@ -0,0 +1,289 @@
...
+
+```shell
+$ git clone https://github.com/pingcap/tidb-operator.git
+$ GOOS=${YOUR_GOOS} make cli
+$ mv tkctl /usr/local/bin/tkctl
+```
+
+## 命令自动补全
+
+你可以配置 `tkctl` 的自动补全来简化使用：
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
你可以配置 `tkctl` 的自动补全以简化使用。
```

**Comment by yikeke** [comment]
I have updated all code blocks in this file to apply the new code block format rules, see details in [Update code block format to better support copy](https://internal.pingcap.net/jira/browse/ICM-95). 
PTAL @aylei

File: tidb-in-kubernetes/tkctl.md [hunk 17]
```diff
@@ -0,0 +1,387 @@
...
+tkctl debug demo-cluster-tikv-0
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+gdb /proc/${pid:-1}/root/tikv-server 1
+```
+
+`tidb-debug` 中预配置的 `.gdbinit` 会将 `sysroot` 设置为 `/proc/1/root/`，因此在 `tidb-debug` 中，假如目标容易的 pid 为 1，则下面的命令可以省略。
```
Comment (yikeke) [inline tidb-in-kubernetes/tkctl.md]: Is "目标容易" a typo?

File: tidb-in-kubernetes/tkctl.md [hunk 18]
```diff
@@ -0,0 +1,387 @@
...
+
+### tkctl debug [pod_name]
+
+该命令用于诊断 TiDB 集群中的 Pod。
+
+实际使用时，该命令会在目标 Pod 的宿主机上以指定镜像启动一个 debug 容器，该容器会与目标 Pod 中的容器共享 namespace，因此可以无缝使用 debug 容器中的各种工具对目标容器进行诊断。
+
+| 参数            | 缩写      | 描述                                                                                                                            |
+| -----           | --------- | -----------                                                                                                                     |
+| --image         |           | 指定 debug 容器使用的镜像，默认为 `pingcap/tidb-debug:lastest`                                                                  |
```
Comment (tennix) [inline tidb-in-kubernetes/tkctl.md]: typo: lastest -> latest

File: tidb-in-kubernetes/tkctl.md [hunk 19]
```diff
@@ -0,0 +1,387 @@
...
+{{< copyable "shell-regular" >}}
+
+```shell
+cp /proc/1/root/tikv-server /
+```
+
+{{< copyable "shell-regular" >}}
+
+```shell
+./run_flamegraph.sh 1
```
Comment (tennix) [inline tidb-in-kubernetes/tkctl.md]: This command will generate an SVG format graph and upload to https://transfer.sh, so users can view or download it from the output URL.
I think this should be noted here.

File: tidb-in-kubernetes/tkctl.md [hunk 20]
```diff
@@ -0,0 +1,387 @@
...
+
+`tkctl` (TiDB Kubernetes Control) 是为 TiDB in Kubernetes 设计的命令行工具，用于运维集群和诊断集群问题。
+
+## 安装
+
+安装 `tkctl` 时，可以直接下载预编译的可执行文件，也可以自行从源码进行编译。
+
+### 下载预编译的可执行文件
+
+- [MacOS](http://download.pingcap.org/tkctl-darwin-amd64-latest.tgz)
```
Comment (tennix) [inline tidb-in-kubernetes/tkctl.md]: Better to change it to `https`.

File: tidb-in-kubernetes/tkctl.md (hunk 18 above)
Comment (aylei) [reply tidb-in-kubernetes/tkctl.md]: Good catch

**Comment by aylei** [comment]
@tennix @yikeke All comments addressed, PTAL again

File: tidb-in-kubernetes/tkctl.md [hunk 21]
```diff
@@ -0,0 +1,389 @@
...
+tkctl get tikv
+```
+
+```
+NAME                  READY   STATUS    MEMORY          CPU   RESTARTS   AGE     NODE
+demo-cluster-tikv-0   2/2     Running   2098Mi/4196Mi   2/2   0          3m19s   172.16.4.155
+demo-cluster-tikv-1   2/2     Running   2098Mi/4196Mi   2/2   0          4m8s    172.16.4.160
+demo-cluster-tikv-2   2/2     Running   2098Mi/4196Mi   2/2   0          4m45s   172.16.4.157
+$ tkctl get volume
+tkctl get volume
```
Comment (weekface) [inline tidb-in-kubernetes/tkctl.md]:
```suggestion
```

Comment (yikeke) [reply tidb-in-kubernetes/tkctl.md]: Updated.

**Comment by yikeke** [comment]
PTAL again. @tennix  @weekface

**Pull request merged**